From: Intiyaz Basha <intiyaz.basha@cavium.com>
Date: Fri, 18 Aug 2017 13:07:19 -0700
Subject: liquidio: fix Smatch error
Patch-mainline: v4.14-rc1
Git-commit: d2896116dbc7be7cabd5db414e008aef4a5e0a00
References: bsc#1067126

Fix Smatch error by not dereferencing iq pointer if it's NULL.

See http://marc.info/?l=kernel-janitors&m=150296723301129&w=2

Also, remove unnecessary parentheses.

Fixes: d314ac222829 ("liquidio: moved liquidio_napi_poll to lio_core.c")
Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Intiyaz Basha <intiyaz.basha@cavium.com>
Signed-off-by: Felix Manlunas <felix.manlunas@cavium.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/cavium/liquidio/lio_core.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/ethernet/cavium/liquidio/lio_core.c
+++ b/drivers/net/ethernet/cavium/liquidio/lio_core.c
@@ -668,8 +668,8 @@ static int liquidio_napi_poll(struct nap
 
 #define MAX_REG_CNT  2000000U
 	/* force enable interrupt if reg cnts are high to avoid wraparound */
-	if (((work_done < budget) && (tx_done)) ||
-	    (iq->pkt_in_done >= MAX_REG_CNT) ||
+	if ((work_done < budget && tx_done) ||
+	    (iq && iq->pkt_in_done >= MAX_REG_CNT) ||
 	    (droq->pkt_count >= MAX_REG_CNT)) {
 		tx_done = 1;
 		napi_complete_done(napi, work_done);
